Data Browser Charter

theathorn opened the pull request:
~~**October 19**: Last call for community reviews~~
~~**October 26**: Last call for oversight reviews~~
**November 15**: Approved during DCP Architecture call

theathorn committed:
Initial revision

brianraymor commented on charters/DataBrowser/charter.md:
```diff
@@ -0,0 +1,67 @@
+
+## Milestones and Deliverables
+* Nov 2018:  Deploy a minimum viable product (MVP) feature set as part of the HCA DCP beta test phase.
+* Q4  2018:  Interface to the Matrix Service.
```
Matrix Service -> **Expression** Matrix Service
Based on the current plan of record, isn't the interface to the expression matrix service included in the Nov 2018 milestone?

theathorn replied:
Agreed.

brianraymor commented on charters/DataBrowser/charter.md:
```diff
@@ -0,0 +1,67 @@
+### Technical Lead
+[Hannes Schmidt](mailto:hannes@ucsc.edu)
+
+## Communication
```
Is there a google group mailing list?

theathorn replied:
Not at the moment. May add later.

tburdett commented on charters/DataBrowser/charter.md:
```diff
@@ -0,0 +1,67 @@
+*Analysis Applications* Third party portals listed in the 'Analyze' section of the Data Portal UI.
+
+## Objectives
+The objective of the Data Browser is to provide an easy-to-use User Interface (UI) allowing users to search, download and analyze data stored in the DCP at petabyte scale on public clouds.
```
This suggests just user interface, whereas the core capabilities describe an API too. Update this to resolve the conflict (I think you probably want to add API here?)

theathorn replied:
Agreed.

tburdett commented on charters/DataBrowser/charter.md:
```diff
@@ -0,0 +1,67 @@
+
+*Analysis Applications* Third party portals listed in the 'Analyze' section of the Data Portal UI.
+
+## Objectives
```
I'll ref [my comment](https://github.com/HumanCellAtlas/dcp-community/pull/10#discussion_r223229320) on the portal charter here.

I want to see an indication somewhere of who is responsible for providing users of data from the DCP a sensible view over that data. Over time, metadata and data will change - in terms of schema, data file formats, QC, analysis results generated from new versions of pipelines. Where in the DCP is this wrangled into a single up-to-date view to provide some guarantees to users over the quality and contemporaneity of the data they get from us?

theathorn replied:
A larger DCP-wide discussion outside the remit of just this charter.

tburdett reviewed (changes requested):
This is mostly clear. I have one UI/API comment that needs to be addressed, otherwise I am happy to approve. I've asked one very big question that needs answering _somewhere_ (but not necessarily here)

malloryfreeberg commented on charters/DataBrowser/charter.md:
```diff
@@ -0,0 +1,67 @@
+The Data Browser (available via the Data Portal 'Explore' section) is an application that provides a faceted search capability allowing users to search and download subsets of cellular resolution data hosted by the Data Coordination Platform (DCP).
+
+## Definitions
+*DCP* The Data Coordination Platform is the name given to the entire system used to ingest, validate, store, analyzes, and make available the data in the Human Cell Atlas project.
```
Replace "analyzes" with "analyze"

theathorn replied:
Done.

theathorn committed:
Minor wording changes.

brianraymor commented on charters/DataBrowser/charter.md:
```diff
@@ -0,0 +1,66 @@
+### Slack Channels
+* HumanCellAtlas/data-browser: data browser development discussions
+
+## Github repositories
```
Could you add a brief description of the repo(s)?

theathorn replied:
Done.

brianraymor committed:
Editorial updates to Slack Channels

Adopting best practice for slack channel formats

theathorn committed:
Add github repos descriptions.

lauraclarke commented on charters/DataBrowser/charter.md:
```diff
@@ -0,0 +1,66 @@
+
+## Description
+The Data Browser (available via the Data Portal 'Explore' section) is an application that provides a faceted search capability allowing users to search and download subsets of cellular resolution data hosted by the Data Coordination Platform (DCP).
+
```
Should this mention the goal of acting like a shopping cart for analysis applications? that isn't evident from this description?

theathorn replied:
Added such a clause to the Description.

lauraclarke commented on charters/DataBrowser/charter.md:
```diff
@@ -0,0 +1,66 @@
+## In-scope
+
+### Core capabilities
+* Create indexes to access files in the Data Store by using public Data Store Service (DSS) APIs to read the metadata.
```
What is the relationship between the Data Browser and the Query service?

theathorn replied:
The Data Browser receives notifications from the Data Store and constructs its own Indexes. It does not use the Query Service.

lauraclarke commented on charters/DataBrowser/charter.md:
```diff
@@ -0,0 +1,66 @@
+* Support multiple metadata schemas via a shared schema access library.
+* Provide multiple search views of the DCP data such as by Projects, Files and Specimens.
+* Support for specifying multiple search criteria ('facets') through the UI.
+* Provide for download of selected files from the Data Store via the UI and/or HCA DCP Command Line Interface (CLI).
```
When you use the word selected here do you mean user selected or that the browser will only enable download of a subset of files?

theathorn replied:
Changed to "user-selected".

lauraclarke commented on charters/DataBrowser/charter.md:
```diff
@@ -0,0 +1,66 @@
+
+### Community engagement
+* Provide triage of feature requests from the community and integration of such requests into the Data Browser roadmap.
+* Outreach and engagement of the community on use/usability of the APIs.
```
Just the APIs or also the UI?

theathorn replied:
Added "UI".

theathorn committed:
Update Description and minor wording changes

Update Description to include hand-off of search results to Matrix Service and Analysis Apps.
Clarify "user-selected files".
Add "UI" to community outreach goals.

theathorn committed:
Add metdata-api to repo list

tburdett reviewed (approved):
Thanks for the changes, this looks good
